tests/rptest/tests/schema_registry_test.py

```diff
@@ -596,3 +599,158 @@ def test_delete_subject_version(self):
         self.logger.debug(result_raw)
         assert result_raw.status_code == requests.codes.ok
         assert result_raw.json() == [1, 2, 3]
+
+    @cluster(num_nodes=3)
+    def test_concurrent_writes(self):
+        schema_template = '{"type":"record","name":"record_%s","fields":[{"type":"string","name":"f1"}]}'
+
+        # How many errors before a worker gives up?
+        max_errs = 1
+
+        # Warm up the servers (schema_registry doesn't create topic etc before first access)
+        for node in self.redpanda.nodes:
+            r = self._request("GET",
+                              "subjects",
+                              hostname=node.account.hostname)
+            assert r.status_code == requests.codes.ok
+
+        class WriteWorker(threading.Thread):
```

Comment:
this stress tester is great, but it's a little problematic to write it this way. any of the python code that you see when writing tests all run on an orchestrator node. the orchestrator node in turn may run lots of tests in parallel, and critically, it may be distant from the cluster (e.g. orchestrate ducktape tests on aws nodes from my laptop).

The strategy to avoid this is to factor out the core of the stress tester into a separate python script (or really any executable you want) that can be run as a shell command. Next, arrange for this helper to be available in the nodes. For python this is generally trivial since most / all of the tree is auto mounted in the container. You can also add whatever dependencies you want in tests/docker/Dockerfile.

Then back in this test harness instead of firing up the background OS thread, sub-class from ducktape.BackgroundThreadedService (or something along these lines) and use `node.ssh_capture` to invoke the helper. Ducktape will automatically allocate a node for the stress tester to run on and arrange for `node` to be whatever hardware it allocated. There are a couple examples in the tree.

Reply:
This is done, although I don't think the source tree is mounted on the test nodes (just the build outputs), so I'm copying the script out at runtime and then removing it in `clean_nodes`.

There's some amount of code duplication as the new standalone workload script gets some of the same http helper methods as the test class has.  If we do more of this kind of thing, a slightly slicker remote python execution capability would come in useful (incl. things like propagating log levels, getting logs back without double-formatting them).

Reply:
thanks. I suspect, and you also pointed out in out-of-band conversation, that we'll be seeing more and more of this so getting a handle on some patterns to make it easier to work with seems critical.